src/hdnode.js

```diff
@@ -239,7 +239,7 @@ HDNode.prototype.derive = function (index) {
 
   // Private parent key -> private child key
   var derivedKeyPair
-  if (this.keyPair.d) {
+  if (!this.isNeutered()) {
```

Comment:
I'm not sure if `isNeutered` is that great after all.
Its hard to derive the intended meaning... `isPrivate` would be much clearer here.

@runn1ng could you add both?
We'll know by `3.0.0` as to whether we should clean it up or not...